### PR TITLE
PixDQMFedOccupancy

### PR DESCRIPTION
**Pull request by fioriNTU**
prevent empty histos in Offline Gui

### CODE REVIEW

**Comment by deguio** [comment]
+1

**Comment by deguio** [comment]
please test

**Comment by davidlange6** [comment]
@fioriNTU  @deguio  - could you summarize the concept of this change? It seems in the "!offline" functionality, an array is reset every event if the lumi section happens to be divisible by 8 or 1 and in the "offline" case it will never by reset.

**Comment by fioriNTU** [comment]
Yes David, is exactly like that. The fact is that in the Online DQM it is ok to monitor the FED occupancy Lumi by Lumi (or onece each 8 Lumis), while in the Offline it makes no sense to have a Monitor Element filled only with data of the the last Lumi, in this context we need a ME representative of the data in the full run. Moreover if, for any reason, the Pixel detector doesn't give data at the end of the run this histogram is empty in the offline DQM and shifters are confused.

**Comment by davidlange6** [comment]
@fioriNTU naively it seems that the code is doing this reset every event (well, for some lumi sections). Is that not the case?

**Comment by fioriNTU** [comment]
@davidlange6 , actually you are right, I was confused by the lumisection%1 statement (that is always true), in the other case (with modOn=true) it does the reset for each event in a lumisection that is a multiple of 8. Probably it need a fix also for the Online case, what do you think?

**Comment by davidlange6** [comment]
Its up to what you want to monitor.. I would assume that you want to reset on the first event of every N lumi sections, but really know the purpose of the plots

**Comment by fioriNTU** [comment]
@danduggan @davidlange6 hey Dan, I guess that the lumisection%1 statement is there because we want to monitor the mean occupancy Lumi by Lumi and not event by event, is that correct?
In the case I will make another small modification and pull request again tomorrow. Let me know.

**Comment by deguio** [comment]
@fioriNTU 
you can also update this PR (and corresponding PRs in 75 and 76).
thanks,
F.

**Comment by deguio** [comment]
-1
updates needed. @danduggan
